test/hotspot/jtreg/serviceability/jvmti/HeapMonitor/MyPackage/HeapMonitor.java

```diff
@@ -198,10 +212,11 @@ public static void freeStorage() {
     if (enableSampling) {
       enableSamplingEvents();
     }
-
+    // similar reason with sampleEverything, consume TLAB and trigger sample
```

Comment:
How about:
`    // Use System.gc() to consume TLAB and trigger sampling as described above in sampleEverything`